tokio/src/time/driver/sleep.rs

```diff
@@ -72,7 +72,9 @@ pub fn sleep_until(deadline: Instant) -> Sleep {
 ///
 /// No work is performed while awaiting on the sleep future to complete. `Sleep`
 /// operates at millisecond granularity and should not be used for tasks that
-/// require high-resolution timers.
+/// require high-resolution timers. The implementation is platform specific
+/// such that the actual resolution may be greater than 1 millisecond and vary
+/// between platforms.
```

Comment:
I would like this to be more specific if possible.

Reply:
Should the docs define the behavior to be expected for each supported platform based on the current implementation ( think it would just link the mio docs)? I found a few issues about the timer granularity/precision, but none that describe configuring on windows:
https://github.com/tokio-rs/tokio/issues/367
https://github.com/tokio-rs/tokio/issues/970

Both those issues lead to a solution using this crate: https://github.com/DoumanAsh/async-timer

Poking around the code a little bit I found the code for the buckets and see why the granularity is limited to 1ms by the tokio runtime. And it looks like the ~16ms resolution for windows comes from the OS events implementation in mio, from the [mio docs](https://docs.rs/mio/latest/mio/struct.Poll.html#implementation-notes) it states `IOCP` is used to communicate with Windows which I think is what limits the resolution. The `async-timer` crate creates a timer directly using a different windows API call.

Reply:
I remember seeing some issue that mentions a call you can make on windows that makes the IOCP timer more precise, but I can't find it...